src/coreclr/src/jit/morph.cpp

```diff
@@ -4262,10 +4262,9 @@ GenTree* Compiler::fgMorphMultiregStructArg(GenTree* arg, fgArgTabEntry* fgEntry
 #if FEATURE_MULTIREG_ARGS
     // Examine 'arg' and setup argValue objClass and structSize
     //
-    CORINFO_CLASS_HANDLE objClass = gtGetStructHandleIfPresent(arg);
-    noway_assert(objClass != NO_CLASS_HANDLE);
-    GenTree* argValue   = arg; // normally argValue will be arg, but see right below
-    unsigned structSize = 0;
+    const CORINFO_CLASS_HANDLE objClass   = gtGetStructHandle(arg);
```

Comment:
This replaces a `noway_assert` with a regular assert; was this intentional? I think it doesn't matter because this path is not optimization-dependent, so a retry after a `noway_assert` would also fail.

Reply:
Yes, it was intentional. I think it is good in general to replace `noway_asserts` that were added as conservative checks for new features in 5.0 while we are in the first stage of 6.0.

Reply:
I think that's reasonable, but we should be thoughtful about removing `noway_asserts` that occur on an optimization-only path, where a retry without optimization could succeed.